### PR TITLE
Anko - Change Arrived Dialog From Xml To Anko

### PR DESCRIPTION
**Pull request by at-datbui**
#### Related issues (Required)
- Task #143

#### Description (Required)
- [x] Change arrived dialog from xml to ank

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/custom/ArrivedDialog.kt [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        mTime = arguments?.getString(TYPE_TIME, "")
+        mDistance = arguments?.getString(TYPE_DISTANCE, "")
+    }
+
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        ui = DialogUI()
+        return ui.createView(AnkoContext.create(context, this))
+    }
+
+    inner class DialogUI : AnkoComponent<DialogArrived> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/ArrivedDialog.kt]: WARNING: Public classes, interfaces and objects require documentation.

File: app/src/main/java/vn/asiantech/way/ui/custom/ArrivedDialog.kt [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                            }
+                        }
+                    }
+                }
+
+                textView {
+                    backgroundResource = R.color.colorGrayLight
+                }.lparams(matchParent, dip(1))
+
+                textView {}.lparams(matchParent, dip(0), LAYOUT_WEIGHT_SMALL)
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/ArrivedDialog.kt]: replace `{}` by `()`

File: app/src/main/java/vn/asiantech/way/ui/custom/ArrivedDialog.kt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        textSize = px2dip(dimen(R.dimen.text_size_normal))
+    }
+
+    private fun ViewManager.tvDescriptionArrivedInfo(string: Int) = textView(string) {
+        textSize = px2dip(dimen(R.dimen.text_size_little))
+    }
+
+    override fun onStart() {
+        super.onStart()
+        val displayMetrics: DisplayMetrics = resources.displayMetrics
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/ArrivedDialog.kt]: Please create a util class for this function

File: app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import android.app.Dialog
+import android.content.Context
+import android.util.DisplayMetrics
+import android.view.WindowManager
+
+/**
+ * Copyright © 2017 Asian Tech Co., Ltd.
+ * Created by datbuit. on 01/12/2017.
+ */
+class ScreenUtil {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt]: add `private constructor` and move all method to `companion object { ... }`

File: app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    companion object {
+        private const val TYPE_UNIT_DIALOG_HEIGHT = 5
+        private const val TYPE_DIALOG_MARGIN_WIDTH = 80
+
+        fun getScreenSize(context: Context, dialog: Dialog) {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt]: WARNING: Public functions require documentation.

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

File: app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ * Copyright © 2017 Asian Tech Co., Ltd.
+ * Created by datbuit. on 01/12/2017.
+ */
+class ScreenUtil private constructor() {
+
+    companion object {
+        private const val TYPE_UNIT_DIALOG_HEIGHT = 5
+        private const val TYPE_DIALOG_MARGIN_WIDTH = 80
+
+        internal fun getScreenSize(context: Context, dialog: Dialog) {
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/utils/ScreenUtil.kt]: Change to extension function

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -29,7 +29,7 @@
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
 
-        <activity android:name=".ui.home.HomeActivity">
```
Comment (at-tienhoang) [inline app/src/main/AndroidManifest.xml]: Remove this change

File: app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt [hunk 8]
```diff
@@ -296,3 +298,7 @@ class BottomButtonCard(context: Context) :
         return this
     }
 }
+
+internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/BottomButtonCard.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 9]
```diff
@@ -22,6 +24,13 @@ internal fun ViewManager.rippleView(init: RippleView.() -> Unit): RippleView {
...
-}
+internal fun Dialog.setDialogScreenSize(context: Context) {
+    val displayMetrics: DisplayMetrics = context.resources.displayMetrics
+    val width = displayMetrics.widthPixels
+    val height = displayMetrics.heightPixels
+    val windowParams = WindowManager.LayoutParams()
+    windowParams.copyFrom(this.window.attributes)
+    windowParams.height = height - height / 5
+    windowParams.width = width - 80
+    this.window.attributes = windowParams
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Checks whether files end with a line separator.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 10]
```diff
@@ -22,6 +24,13 @@ internal fun ViewManager.rippleView(init: RippleView.() -> Unit): RippleView {
...
-internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
-    return ankoView({ BottomButtonCard(it) }, 0, init)
-}
+internal fun Dialog.setDialogScreenSize(context: Context) {
+    val displayMetrics: DisplayMetrics = context.resources.displayMetrics
+    val width = displayMetrics.widthPixels
+    val height = displayMetrics.heightPixels
+    val windowParams = WindowManager.LayoutParams()
+    windowParams.copyFrom(this.window.attributes)
+    windowParams.height = height - height / 5
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 11]
```diff
@@ -22,6 +24,13 @@ internal fun ViewManager.rippleView(init: RippleView.() -> Unit): RippleView {
...
+    windowParams.width = width - 80
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

**Pull request merged**